tagstudio/src/qt/modals/tag_search.py

```diff
@@ -153,6 +157,73 @@ def update_tags(self, query: str = ""):
             l.addWidget(ab)
             self.scroll_layout.addWidget(c)
 
+        # Add a create tag button if a query is entered
+        if query:
+            c = self.create_tag_button(query)
+            self.scroll_layout.addWidget(c)
+
+        self.search_field.setFocus()
+
+    def create_tag_button(self, query: str):
+        # Construct the create tag button
+        c = QWidget()
+        l = QHBoxLayout(c)
+        l.setContentsMargins(0, 0, 0, 0)
+        l.setSpacing(3)
+
+        create_button = QPushButton(self)
+        create_button.setFlat(True)
+        create_button.setText(f"Create \"{query.replace("&", "&&")}\"")
+
+        inner_layout = QHBoxLayout()
+        inner_layout.setObjectName("innerLayout")
+        inner_layout.setContentsMargins(2, 2, 2, 2)
+        create_button.setLayout(inner_layout)
+        create_button.setMinimumSize(math.ceil(22 * 1.5), 22)
+
+        create_button.setStyleSheet(
+            f"QPushButton{{"
+            f"background: {get_tag_color(ColorType.PRIMARY, "dark gray")};"
+            f"color: {get_tag_color(ColorType.TEXT, "dark gray")};"
+            f"font-weight: 600;"
+            f"border-color:{get_tag_color(ColorType.BORDER, "dark gray")};"
+            f"border-radius: 6px;"
+            f"border-style:solid;"
+            f"border-width: {math.ceil(1*self.devicePixelRatio())}px;"
+            f"padding-right: 4px;"
+            f"padding-bottom: 1px;"
+            f"padding-left: 4px;"
+            f"font-size: 13px"
+            f"}}"
+            f"QPushButton::hover{{"
+            f"border-color:{get_tag_color(ColorType.LIGHT_ACCENT, "dark gray")};"
+            f"}}"
+        )
+
+        create_button.clicked.connect(lambda x=query: self.create_tag(query))
+        l.addWidget(create_button)
+
+        return c
+
+    def create_tag(self, query: str):
+        # Fuction called when tag should be created based on query
+        # Creates the tag and applies it to the document
+        new_tag: Tag = Tag(
+            id=-2,
```

Comment:
wont this cause a collision among multiple tags created this way? Or is the ID somehow reassigned to different value elsewhere in the code?